### DISCUSSION TITLE
How to prepare a multi-speaker dataset?

### DISCUSSION THREAD

**Discussion by rikochet9** [rhasspy/piper#575] [Q&A]
I'm looking into how to train a multi-speaker model as per this documentation: https://github.com/rhasspy/piper/blob/master/TRAINING.md

Under Preparing a Dataset, it says to have the metadata.csv file include speaker names on each line. Is there some way to set Piper Recording Studio to include them in this file or am I meant to go into the file and manually specify the speaker name for each line?

![image](https://github.com/user-attachments/assets/3ba16476-1573-42fd-810c-d6f9cb9597a3)

**Comment by StoryHack** [comment]
Isn't there a way to set up mutliple users in the recording studio? I would think the speaker/user would be tracked automatically and added in.

**Reply by rikochet9** [reply]
Yes, it supports multiple users but it's users in the sense of different people using the same computer. What it does is store each user's recordings completely separately.

**Reply by StoryHack** [reply]
If you can get them out separately as their own datasets, then it's not too hard to edit the metadata.csv for each one then merge them. Use a text editor that has a search and replace function to search for the "|" symbol and replace it with "|nameofvocie|". I don't remember how it names the wav files, but you might have to do a bulk renaming on them to add something unique (and updating the metadata file, too) in order to be able to throw them in the same directory with the other speaker's recordings. 

I've had to do that when merging datasets before. The easiest way was to add a "_nameofvoice" in the filenames so "0001.wav" became "0001_nameofvoice.wav" then I replaced "|" with "_nameofvoice|nameofvoice|" everywhere it appears in the metadata.csv file. I had scripted my own thing that was copying and renaming the wavs, but there are many bulk/batch renaming utilities available. 

It's a bit a pain, to be sure, but a million times easier than doing it one at a time. And the recording studio may already spit out unique file names that don't need to be changed.

**Reply by rikochet9** [reply]
Thanks, that sounds viable. I'll give it a try on my next go!
